docs/integrations/sources/mongodb-v2.md

```diff
@@ -160,10 +160,28 @@ The MongoDB source utilizes change data capture (CDC) as a reliable way to keep
 
 Airbyte utilizes [the change streams feature](https://www.mongodb.com/docs/manual/changeStreams/) of a [MongoDB replica set](https://www.mongodb.com/docs/manual/replication/) to incrementally capture inserts, updates and deletes using a replication plugin. To learn more how Airbyte implements CDC, refer to [Change Data Capture (CDC)](https://docs.airbyte.com/understanding-airbyte/cdc/).
 
+### No Schema Mode
+
+When the schema enforced option is disabled, MongoDB collections are read in schema-less mode which doesn't assume documents share the same structure.
+This allows for greater flexibility in reading data that is unstructured or vary a lot in between documents in a single collection.
+In No Schema mode each document will generate a record that only contains the following fields:
+
+```json
+{
+  "_id": <document id>,
+  "data": {<a JSON cotaining the entire set of fields found in document>}
+}
+```
+The contents of `data` will vary according to whichever values exist on each document read from MongoDB.
+Unlike in Schema enforced mode, the same field can vary in type between document. For example field `"xyz"` may be a String on one document and a Date on another. 
+As a result no field will be omitted and no document will be rejected.
+
+
 ## Limitations & Troubleshooting
 
 * Only supports [replica set](https://www.mongodb.com/docs/manual/replication/) cluster type.
-* Schema discovery uses [sampling](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) of the documents to collect all distinct top-level fields.  This value is universally applied to all collections discovered in the target database.  The approach is modelled after [MongoDB Compass sampling](https://www.mongodb.com/docs/compass/current/sampling/) and is used for efficiency.  By default, 10,000 documents are sampled.  This value can be increased up to 100,000 documents to increase the likelihood that all fields will be discovered.  However, the trade-off is time, as a higher value will take the process longer to sample the collection.
+* Schema discovery uses [sampling](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sample/) of the documents to collect all distinct top-level fields.  This value is universally applied to all collections discovered in the target database.  The approach is modelled after [MongoDB Compass sampling](https://www.mongodb.com/docs/compass/current/sampling/) and is used for efficiency.  By default, 10,000 documents are sampled.  This value can be increased up to 100,000 documents to increase the likelihood that all fields will be discovered.  However, the trade-off is time, as a higher value will take the process longer to sample the collection. 
+* When Running with Schema Enforced set to `false` there is no attempt to discover any schema. See more in [No Schema Mode](#No-Schema-Mode).
```

Comment:
Should we reword this to say that you can't use column selection? If I understand correctly, that's the customer impact here.

Reply:
I'll add something on column selection.
I filed #32170 to deal with it (not an MVP for schemaless)

Reply:
I added something on lack of column selection in the schema enforecement section

I filed #32170 to deal with field de/selection in schemaless mode by the way